flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/nodes/datastream/DataStreamWindowJoin.scala

```diff
@@ -87,44 +98,38 @@ class DataStreamRowStreamJoin(
 
     val config = tableEnv.getConfig
 
+    val leftIsAccRetract = DataStreamRetractionRules.isAccRetract(left)
```

Comment:
Retractions are just one (of two) ways to encode updates in a stream. This join implementation does not support any kind of updates (which is fine). However, we have to check if the input streams produce updates or not. The `StreamTableEnvironment` has a private class `AppendOnlyValidator`. We should move `AppendOnlyValidator` and `UniqueKeyExtractor` into a class `UpdateUtils` together with the methods `isAppendOnly()` and `getUniqueKeyFields()`. `isAppendOnly()` checks if an operator produces updates or not and should be used here.